cadc-gms/src/main/java/org/opencadc/auth/StandardIdentityManager.java

```diff
@@ -132,19 +129,15 @@ public class StandardIdentityManager implements IdentityManager {
     public StandardIdentityManager() {
         LocalAuthority loc = new LocalAuthority();
         String key = Standards.SECURITY_METHOD_OPENID.toASCIIString();
-        this.oidcIssuer = loc.getServiceURI(key);
-        try {
-            URL u = oidcIssuer.toURL();
-            oidcDomains.add(u.getHost());
+        oidcClient = new OIDCClient(loc.getServiceURI(key));
 
-            // add known and assume trusted A&A services
-            String host = getProviderHostname(loc, Standards.GMS_SEARCH_10);
-            if (host != null) {
-                oidcDomains.add(host);
-            }
+        URL u = oidcClient.getIssuerURL();
```

Comment:
This is where the call to get well know config happens and it occurs for every new IM instance. **That is going to be costly** - previous impl did zero calls on instantiation (well, reg calls but they are cached on local machine for ~10 min) and then probably one userinfo call and optionally one posix-mapper call later on.

For reasons, the AuthenticationUtil.getIdentityManager() call that instantiates an IM can happen from multiple places within a request and each of those will suffer this overhead.
---
I'm wondering if OIDCClient can re-use CachingFile from cadc-registry to cache the well-known config in the filesystem... something like $TMP/{user}/cadc-gms/{issuer hostname}/{path from url}